### ISSUE TITLE
koa server always responds 404

### ISSUE THREAD

**Issue by mingyang91** [pleerock/routing-controllers#210]
In koa async controller function, if I set `@Authroized()` , server always responds `"404 Not Found"`

I think this code Incorrect
```
const handleError = (result: any) => {
    if (!result) {
        let error = actionMetadata.authorizedRoles.length === 0 ? new AuthorizationRequiredError(action) : new AccessDeniedError(action);
        return this.handleError(error, actionMetadata, action);
    } else {
        next();
    }
};

if (isPromiseLike(checkResult)) {
    checkResult.then(result => handleError(result));
} else {
    handleError(checkResult);
}
```
https://github.com/pleerock/routing-controllers/blob/master/src/driver/koa/KoaDriver.ts#L101-L108

it should be modified as follows.
```
const handleError = (result: any) => {
    if (!result) {
        let error = actionMetadata.authorizedRoles.length === 0 ? new AuthorizationRequiredError(action) : new AccessDeniedError(action);
        return this.handleError(error, actionMetadata, action);
    } else {
        return next();
    }
};

if (isPromiseLike(checkResult)) {
    return checkResult.then(result => handleError(result));
} else {
    return Promise.resolve(handleError(checkResult));
}
```
https://github.com/mingyang91/routing-controllers/blob/master/src/driver/koa/KoaDriver.ts#L107-L114

**Comment by 19majkel94** [comment]
Could you make a PR and test the changes? 😉

**Comment by smithdalec** [comment]
Same is happening for me using koa@2 with a non-async controller action decorated with `@Authorized()`. The response gets sent back with a 404 status before the controller action even runs.

**Comment by 19majkel94** [comment]
@mingyang91 Will you find some time to make a PR as you described the problem and found the fix:
https://github.com/mingyang91/routing-controllers/commit/38a68e6fa1ef73439c8ca7c40225608a9be62aa3#diff-112df671cfc9c4c162793234f846b6d0R112

**Comment by mingyang91** [comment]
sure, i will make a pr in next week.

**Comment by NoNameProvided** [comment]
This is fixed in #240. Will be released in `0.7.2`.

**Comment by 19majkel94** [comment]
Ok, so I'm closing this issue. @mingyang91 feel free to reopen if the issue will still exist in 0.7.2.
